# Logistic Regression Example

Simple demonstration of the Logistic Regression model

In [3]:
# Importing useful libraries
import numpy as np
#import matplotlib.pyplot as plt
import pandas as pd

In [4]:
# Importing the dataset
default = pd.read_csv('../data/credit_card_default.csv', index_col="ID")
# Renaming columns
default.rename(columns=lambda x: x.lower(), inplace=True)
default = default[['pay_amt1','age','default payment next month']]
default.rename(columns={'default payment next month':'default'}, inplace=True)
default['pay_amt1'] = default['pay_amt1']/1000

In [ ]:
default.head()

### Building a Logistic Regression model with two features: Limit Balance and Age
 
Positive class: Default(==1)
$$Pr(default=1 | X) = \frac{1}{1+exp(-Z)}$$
Where: 
$$Z = w_0 + w_1 LimitBalance + w_2 Age$$

In [5]:
X = default[['pay_amt1','age']]
y = default['default']

In [6]:
# 1. Import the estimator object (model)
from sklearn.linear_model import LogisticRegression

# 2. Create an instance of the estimator
logreg = LogisticRegression()

# 3. Use the trainning data to train the estimator
logreg.fit(X, y)

LogisticRegression(C=1.0, class_weight=None, dual=False, fit_intercept=True,
          intercept_scaling=1, max_iter=100, multi_class='ovr', n_jobs=1,
          penalty='l2', random_state=None, solver='liblinear', tol=0.0001,
          verbose=0, warm_start=False)

In [7]:
print(logreg.intercept_)
print(logreg.coef_)

[-1.26020233]
[[-0.03658608  0.00444121]]


$$Pr(default=1 | X) = \frac{1}{1+exp(-Z)}$$

Where: 

$$Z = -1.2602 -0.036586 \times LimitBalance + 0.004441 \times Age$$

In [8]:
def Z(pay_amt1, age):
    return logreg.intercept_[0] + logreg.coef_[0][0]*pay_amt1 + logreg.coef_[0][1]*age

def prob_default(limit_balance, age):
    z = Z(limit_balance, age) 
    return 1/(1 + np.exp(-z))

## Predicting probability of default

In [9]:
prob_default(limit_balance=2.0, age=40)

0.23944476149429605